patientsearch/src/js/components/Info.js

```diff
@@ -1,186 +1,220 @@
-            setInitialized(true);
-            setLoading(false);
-        });
-    }, [initialized]);
+  React.useEffect(() => {
     /*
-     * return site specific intro HTML text for the site, e.g. HTML block 1
-     */
-    function getSiteLandingIntroText() {
-        if (!Object.keys(setting)) return "";
-        return setting["LANDING_INTRO"];
-    }
-     /*
-     * return site specific button text for the site
+     * retrieve setting information
      */
-     function getSiteLandingButtonText() {
-        if (!Object.keys(setting)) return "";
-        return setting["LANDING_BUTTON_TEXT"];
-    }
-    /*
-     * return site specific body HTML text for the site, e.g. HTML block 2
-     */
-    function getSiteLandingBodyText() {
-        if (!Object.keys(setting)) return "";
-        return setting["LANDING_BODY"];
-    }
-    function getSiteId() {
-        if (!Object.keys(setting)) return "";
-        return setting[SITE_ID_STRING];
-    }
-    function getSystemType() {
-        if (!Object.keys(setting)) return "test";
-        return setting[SYSTEM_TYPE_STRING];
-    }
-    function handleImageLoaded(e) {
-        if (!e.target) {
-          return false;
-        }
-        let imageLoaded = imageOK(e.target);
-        if (!imageLoaded) {
-          e.target.setAttribute("disabled", true);
-          return;
-        }
-        let defaultLogoImage = document.querySelector(".default-logo");
-        if (defaultLogoImage) {
-          defaultLogoImage.setAttribute("disabled", true);
+    sendRequest("./settings").then(
+      (response) => {
+        let data = null;
+        try {
+          data = JSON.parse(response);
+        } catch (e) {
+          console.log("error parsing data ", e);
         }
+        setSetting(data);
+        setSiteID(getSiteId());
+        setInitialized(true);
+        setTimeout(() => setLoading(false), 250);
+      },
+      (error) => {
+        console.log("Failed to retrieve data", error.statusText);
+        setInitialized(true);
+        setLoading(false);
+      }
+    );
+  }, [initialized]);
+  /*
+   * return site specific intro HTML text for the site, e.g. HTML block 1
+   */
+  function getSiteLandingIntroText() {
+    if (!Object.keys(setting)) return "";
+    return setting["LANDING_INTRO"];
+  }
+  /*
+   * return site specific button text for the site
+   */
+  function getSiteLandingButtonText() {
+    if (!Object.keys(setting)) return "";
+    return setting["LANDING_BUTTON_TEXT"];
+  }
+  /*
+   * return site specific body HTML text for the site, e.g. HTML block 2
+   */
+  function getSiteLandingBodyText() {
+    if (!Object.keys(setting)) return "";
+    return setting["LANDING_BODY"];
+  }
+  function getSiteId() {
+    if (!Object.keys(setting)) return "";
+    return setting[SITE_ID_STRING];
+  }
+  function getSystemType() {
+    if (!Object.keys(setting)) return "test";
+    return setting[SYSTEM_TYPE_STRING];
+  }
```

Comment:
Would it make sense to refactor these into a single function (eg `getConfig(setting_name){}`)?